src/path_rewriting.rs

```diff
@@ -45,72 +45,57 @@ pub fn rewrite_paths(
-            if let Some(p) = path_mapping.get(to_lowercase_first(path.to_str().unwrap())) {
-                (PathBuf::from(p.as_str().unwrap()), true)
-            } else if let Some(p) = path_mapping.get(to_uppercase_first(path.to_str().unwrap())) {
-                (PathBuf::from(p.as_str().unwrap()), true)
-            } else if path.starts_with(&prefix_dir) {
-                (path.strip_prefix(&prefix_dir).unwrap().to_path_buf(), false)
-            } else if path.starts_with(&source_dir) {
-                (path.strip_prefix(&source_dir).unwrap().to_path_buf(), false)
-            } else {
-                (path, false)
-            };
+        let mut rel_path = PathBuf::from(path.replace("\\", "/"));
+
+        // Get path from the mapping.
+        if let Some(ref path_mapping) = path_mapping {
+            if let Some(p) = path_mapping.get(to_lowercase_first(rel_path.to_str().unwrap())) {
+                rel_path = PathBuf::from(p.as_str().unwrap());
+            } else if let Some(p) = path_mapping.get(to_uppercase_first(rel_path.to_str().unwrap())) {
+                rel_path = PathBuf::from(p.as_str().unwrap());
+            }
+        }
+
+        // Remove prefix from path.
+        if let Some(ref prefix_dir) = prefix_dir {
+            if rel_path.starts_with(&prefix_dir) {
+               rel_path = rel_path.strip_prefix(&prefix_dir).unwrap().to_path_buf()
+            }
+        }
+
+        let mut abs_path = rel_path.clone();
 
         // Get absolute path to source file.
-        let abs_path = if rel_path.is_relative() {
-            if !cfg!(windows) {
-                PathBuf::from(&source_dir).join(&rel_path)
-            } else {
-                PathBuf::from(&source_dir).join(&rel_path.to_str().unwrap().replace("/", "\\"))
+        if let Some(ref source_dir) = source_dir {
+            if rel_path.is_relative() {
+                if !cfg!(windows) {
+                    abs_path = PathBuf::from(&source_dir).join(&rel_path);
+                } else {
+                    abs_path = PathBuf::from(&source_dir).join(&rel_path.to_str().unwrap().replace("/", "\\"));
+                }
             }
-        } else {
-            rel_path.clone()
-        };
+        }
 
         // Canonicalize, if possible.
-        let abs_path = match canonicalize_path(&abs_path) {
-            Ok(p) => p,
-            Err(_) => abs_path,
-        };
+        if let Ok(p) = canonicalize_path(&abs_path) {
+            abs_path = p;
+        }
 
-        let rel_path = if found_in_mapping {
-            rel_path
-        } else if abs_path.starts_with(&source_dir) {
-            // Remove source dir from path.
-            abs_path.strip_prefix(&source_dir).unwrap().to_path_buf()
-        } else {
-            abs_path.clone()
-        };
+        if let Some(ref source_dir) = source_dir {
+            // Now the path could be different (if there was a symlink)
+            if abs_path.starts_with(&source_dir) {
```

Comment:
if source_dir is relative or if it's not canonicalized and abs_path is absolute then the condition should always be false

Reply:
but it won't never happen since we always pass a canonical path for source_dir (cf main.rs)